Source/Urho3D/IO/File.cpp

```diff
@@ -148,26 +142,26 @@ bool RawFile::Open(const ea::string& fileName, FileMode mode, const PackageEntry
         return false;
     }
 
-    if (packageEntry)
+    offset_ = offset;
+
+    fseek((FILE*)handle_, 0, SEEK_END);
+    long size = ftell((FILE*)handle_);
+    if (size < offset_)
     {
-        size_ = packageEntry->size_;
-        offset_ = packageEntry->offset_;
-        fseek((FILE*)handle_, offset_, SEEK_SET);
+        URHO3D_LOGERRORF("Could not open file %s with offset beyond file size", fileName.c_str());
+        Close();
+        size_ = 0;
+        return false;
     }
-    else
+    if (size > M_MAX_UNSIGNED)
     {
-        fseek((FILE*)handle_, 0, SEEK_END);
-        long size = ftell((FILE*)handle_);
-        fseek((FILE*)handle_, 0, SEEK_SET);
-        if (size > M_MAX_UNSIGNED)
-        {
-            URHO3D_LOGERRORF("Could not open file %s which is larger than 4GB", fileName.c_str());
-            Close();
-            size_ = 0;
-            return false;
-        }
-        size_ = (unsigned)size;
+        URHO3D_LOGERRORF("Could not open file %s which is larger than 4GB", fileName.c_str());
```

Comment:
If we are reworking these parts maybe we should lift this limitation?

Reply:
Would be great, actually but I was trying to avoid it to keep scope of the change somewhat limited. The PR is couple weeks old and I would like to complete it in a foreseeable future